lightning/src/ln/channel.rs

```diff
@@ -8438,19 +8665,33 @@ impl<SP: Deref> InboundV2Channel<SP> where SP::Target: SignerProvider {
 			&context.get_counterparty_pubkeys().revocation_basepoint);
 		context.channel_id = channel_id;
 
-		let chan = Self {
+		let mut channel = Self {
 			context,
 			unfunded_context: UnfundedChannelContext { unfunded_channel_age_ticks: 0 },
 			dual_funding_context: DualFundingChannelContext {
 				our_funding_satoshis: funding_satoshis,
 				their_funding_satoshis: msg.common_fields.funding_satoshis,
-				funding_tx_locktime: msg.locktime,
+				funding_tx_locktime: LockTime::from_consensus(msg.locktime),
 				funding_feerate_sat_per_1000_weight: msg.funding_feerate_sat_per_1000_weight,
+				our_funding_inputs: funding_inputs,
 			},
 			interactive_tx_constructor: None,
 		};
 
-		Ok(chan)
+		let tx_msg_opt_res = channel.begin_interactive_funding_tx_construction(
+			entropy_source,
+		);
+		let pending_msg_send_event = match tx_msg_opt_res {
+			Ok(tx_msg_opt) => tx_msg_opt.map(|tx_msg| tx_msg.into_msg_send_event(&channel.context.counterparty_node_id)),
+			Err(_) => {
+				return Err(ChannelError::Close((
+					"V2 channel rejected due to sender error".into(),
+					ClosureReason::HolderForceClosed { broadcasted_latest_txn: Some(false) },
+				)));
+			}
+		};
```

Comment:
Instead of returning the message to send here, I wonder if it would be cleaner if the call site directly called a method similar to `accept_inbound_dual_funded_channel` to obtain the message. That way the logic for sending the message is only applicable to the V2 case.

Currently, the assumption for V1 is that the newly created channel is always tupled with `None` at the call site. This would affect the return type of `InteractiveTxConstructor::new`, but I think that should be ok? Let me know if I'm missing something.

Also, IIUC, won't we only have a message to send if we are the initiator? (i.e., for inbound channels this will always be `None`)

Reply:
Yes no message if we are acceptor. I suspect this was the result of over zealous copy pasting on my end.

Reply:
I will keep the the former part of this comment in mind for outbound V2 channels.